lib/Github/Client.php

```diff
@@ -146,27 +104,26 @@ class Client
     /**
      * Instantiate a new GitHub client.
      *
-     * @param HttpClient|null $httpClient
-     * @param string|null     $apiVersion
-     * @param string|null     $enterpriseUrl
+     * @param Builder|null $httpClient
+     * @param string|null  $apiVersion
+     * @param string|null  $enterpriseUrl
      */
-    public function __construct(HttpClient $httpClient = null, $apiVersion = null, $enterpriseUrl = null)
+    public function __construct(Builder $httpClientBuilder = null, $apiVersion = null, $enterpriseUrl = null)
```

Comment:
My idea is that client should always be constructed without extra manipulations, i.e. 

```
$github = new Client();
$github->doAnything();
```

maybe static factory can be a possible solution.